ckanext/validation/jobs.py

```diff
@@ -93,9 +93,14 @@ def run_validation_job(resource):
     if schema.get("transpose"):
         altered_df = _transpose_dataframe(original_df)
 
+    # Ingest NA values
+    if schema.get("naValue"):
+        altered_df = _remove_na(schema["naValue"], altered_df)
+
     # Foreign keys requires using resource metadata in validation step
     # We insert the metadata into schema here
-    _prep_foreign_keys(dataset, schema, resource, altered_df)
+    if schema.get('foreignKeys', False):
```

Comment:
`if schema.get('foreignKeys')` is enough here. `None` is interpreted as `False`.